File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

// prints pyramid with a height from 1-8
void pyramid(int n);

int main(void)
{
    // gets users input for the height of the pyramid
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    pyramid(height);
    return 1;
}

void pyramid(int n)
{
    for (int i = 0; i < n; i++)
    {
        // prints spaces before the #
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }
        // prints #
        for (int j = n - i - 1; j < n; j++)
        {
            printf("#");
        }
        // space in the middle
        printf("  ");
        // left part of the pyramid
        for (int j = n - i - 1; j < n; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: pset6/marioMore/mario.py
from cs50 import get_int


def main():
    # prompts user for input that is a number between 1 and 8 inclusive
    while True:
        height = get_int("Height: ")
        if (height < 9 and height > 0):
            break
    print_pyramid(height)


def print_pyramid(height):
    # prints pyramid for specified height
    for i in range(height):
        # prints spaces
        print(" " * (height - 1 - i), end="")
        # prints first #
        print("#" * (i + 1), end="")
        # prints spaces between #
        print("  ", end="")
        #prints second #
        print("#" * (i + 1))


if __name__ == "__main__":
    main()